m2cgen/interpreters/interpreter.py

```diff
@@ -228,3 +228,51 @@ def _cache_reused_expr(self, expr, expr_result):
         self._cached_expr_results[expr] = CachedResult(
             var_name=var_name, expr_result=None)
         return var_name
+
+
+class FunctionalToCodeInterpreter(ToCodeInterpreter):
+    """
+    This interpreter provides default implementation for the methods
+    interpreting AST expression into code.
+
+    It can be used for the most programming languages and requires only
```

Comment:
This paragraph doesn't seem to be accurate for this base class.

Reply:
Done in 464f285c718e5077fea6f9aaaa9db3131867e741.
Added the word `functional`.